yugabyte_cloudformation.yaml

```diff
@@ -0,0 +1,490 @@
+  PublicSubnetRouteTableAssociation1:
+    Type: 'AWS::EC2::SubnetRouteTableAssociation'
+    Properties:
+      SubnetId: !Ref PublicSubnet1
+      RouteTableId: !Ref PublicRouteTable
+  PublicSubnetRouteTableAssociation2:
+    Type: 'AWS::EC2::SubnetRouteTableAssociation'
+    Properties:
+      SubnetId: !Ref PublicSubnet2
+      RouteTableId: !Ref PublicRouteTable
+  YugaByteNodeSG:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: 'YugaByte Node Security Group' 
+      GroupName: 'YugaByteNodeSG'
+      VpcId: !Ref VPC
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 22
+          ToPort: 22
+          CidrIp: "0.0.0.0/0"
+        - IpProtocol: tcp
+          FromPort: 7000
+          ToPort: 7000
+          CidrIp: "0.0.0.0/0"
+        - IpProtocol: tcp
+          FromPort: 7100
+          ToPort: 7100
+          CidrIp: !FindInMap
+            - SubnetConfig
+            - VPC
+            - CIDR
+        - IpProtocol: tcp
+          FromPort: 9000
+          ToPort: 9000
+          CidrIp: "0.0.0.0/0"
+        - IpProtocol: tcp
+          FromPort: 9100
+          ToPort: 9100
+          CidrIp: !FindInMap
+            - SubnetConfig
+            - VPC
+            - CIDR
+        - IpProtocol: tcp
+          FromPort: 6379
+          ToPort: 6379
+          CidrIp: !FindInMap
+            - SubnetConfig
+            - VPC
+            - CIDR
+        - IpProtocol: tcp
+          FromPort: 5433
+          ToPort: 5433
+          CidrIp: !FindInMap
+            - SubnetConfig
+            - VPC
+            - CIDR
+        - IpProtocol: tcp
+          FromPort: 9042
+          ToPort: 9042
+          CidrIp: !FindInMap
+            - SubnetConfig
+            - VPC
+            - CIDR
+      Tags:
+        - Key: 'Name'
+          Value: 'YugaByteSecurityGroup'
+
+  Node0NetInt:
+      Type: 'AWS::EC2::NetworkInterface'
+      Properties:
+        SubnetId: !Ref PublicSubnet0
+        GroupSet:
+          - !Ref YugaByteNodeSG
+  
+  Node1NetInt:
+      Type: 'AWS::EC2::NetworkInterface'
+      Properties:
+        SubnetId: !Ref PublicSubnet1
+        GroupSet:
+          - !Ref YugaByteNodeSG
+  
+  Node2NetInt:
+      Type: 'AWS::EC2::NetworkInterface'
+      Properties:
+        SubnetId: !Ref PublicSubnet2
+        GroupSet:
+          - !Ref YugaByteNodeSG
+
+
+  YugaByteNode0:
+    Type: 'AWS::EC2::Instance'
+    DependsOn: PublicRoute
+    CreationPolicy:
+      ResourceSignal:
+        Count: '1'
+        Timeout: PT5M
+    Metadata:
+       'AWS::CloudFormation::Init':
+        configSets:
+          InstallAndRun:
+            - Install
+            - Configure
+        Install:
+          packages:
+            yum:
+              git: []
+        Configure:
+          files:
+            /home/ec2-user/install_software.sh:
+              source: "https://raw.githubusercontent.com/vp15591/yugabytedb-cloudformation/master/scripts/install_software.sh"
```

Comment:
we need to refer to the install script via relative paths?

Reply:
Yes, File resource in CloudFormation takes a relative path only to create it.